### ISSUE TITLE
Webpack, AMD, CommonJS... which one are you using?

### ISSUE THREAD

**Issue by lamuertepeluda** [SINTEF-9012/PruneCluster#115]
I'm having troubles importing PruneCluster into my webpack project. None of these is working! Always undefined. It seems like the library does not follow any convention, while Leaflet does.

```javascript
//ES5 style
var PruneCluster = require("prunecluster/dist/PruneCluster.js");
//ES6 with Babel
import {PruneCluster, PruneClusterForLeaflet} from "prunecluster/dist/PruneCluster.js";
//hoping to find some default
import "prunecluster/dist/PruneCluster.js";
```

Leaflet
```javascript
import L from "leaflet/dist/leaflet-src.js";
```

I guess you should do something about it...

**Comment by lamuertepeluda** [comment]
By the way a found a solution by myself but it's more an hack that puts the issue in evidence:

```javascript
import L from "leaflet/dist/leaflet-src.js";

var PruneCluster = require("imports?L=leaflet/dist/leaflet-src.js!exports?PruneCluster!prunecluster/dist/PruneCluster.js");
```

I found out this by reading [this docs](https://webpack.github.io/docs/shimming-modules.html).

However there would be no need if you stick to some standard module definitions.

**Comment by yellowiscool** [comment]
Hi,

`module.exports` or `define` are not supported and undefined on web browsers, as far as I know. If you look at Leaflet's source code, [it contains specific code to support AMD or CommonJS module definitions](https://github.com/Leaflet/Leaflet/blob/3e636d26b96b32037752c406709610c7047c8094/src/Leaflet.js).

We could add support for AMD or CommonJS the same way Leaflet does, a PR is welcome.

**Comment by lamuertepeluda** [comment]
That's correct: there is some additional code required to support module definition, just like Leaflet does. However there are obvious advantages when integrating the library in a wider project.

Also [UMD](https://github.com/umdjs/umd/blob/master/README.md) is an option.

I'll see what I can do, given that I'm not familiar with TypeScript at all, in order to make a PR.

**Comment by danvirsen** [comment]
Any updates on this?

**Comment by yellowiscool** [comment]
It's on the TODO list.

**Comment by gtanguay** [comment]
Using [webpack export loader,](https://github.com/webpack-contrib/exports-loader) add this line to your javascript.

`import { PruneCluster, PruneClusterForLeaflet } from 'exports-loader?PruneCluster,PruneClusterForLeaflet!prunecluster/dist/PruneCluster.js'`

**Comment by disarticulate** [comment]
@gtanguay may not be obvious to some:
```
npm install exports-loader --save-dependency
```

**Comment by pedroetb** [comment]
Any update on this? I have been using the fork proposed at #161 pull request since 2018.

**Comment by fungiboletus** [comment]
Hei @pedroetb This project has been on pause for a few years, mostly because I left the company and no one took over.

I hope to resume the development at some point and modernize it, but I cannot give a timeline because I don't have a budget for that yet.

Things has changed with now MapBoxGLJS/MapLibre or Deck.gl. Meanwhile you can continue to use the fork if it still work.
